C standard library <time.h> functions and structures not available when using POSIX API

**Describe the bug**
C standard library time.h functions and structures are not available when using CONFIG_POSIX_API=y

**To Reproduce**
Steps to reproduce the behavior:
1. cd samples/posix/gettimeofday
2. modify src/main:
```
#include <time.h>

// ...

void main(void)
{
    time_t now = time(NULL);
    struct tm tm_now;
    localtime_r(&now, &tm_now);
    printf("Time: %s\n", asctime(&tm_now));
    // ...
}
```
3. west build -b mimxrt1064_evk
4. See error and warnings (struct tm is of unknown size and implicit declarations)

**Expected behavior**
Functions and structures are correctly defined and no error or warning shows up.

**Impact**
I am trying to port a library to Zephyr and I need to use both posix time functions and C time functions.

**Screenshots or console output**
```
../src/main.c: In function 'main':
../src/main.c:18:15: warning: implicit declaration of function 'time' [-Wimplicit-function-declaration]
   18 |  time_t now = time(NULL);
      |               ^~~~
../src/main.c:19:12: error: storage size of 'tm_now' isn't known
   19 |  struct tm tm_now;
      |            ^~~~~~
../src/main.c:20:2: warning: implicit declaration of function 'localtime_r' [-Wimplicit-function-declaration]
   20 |  localtime_r(&now, &tm_now);
      |  ^~~~~~~~~~~
../src/main.c:22:23: warning: implicit declaration of function 'asctime' [-Wimplicit-function-declaration]
   22 |  printf("time: %s\n", asctime(&tm_now));
      |                       ^~~~~~~
../src/main.c:19:12: warning: unused variable 'tm_now' [-Wunused-variable]
   19 |  struct tm tm_now;
      |            ^~~~~~
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
OS: Linux Ubuntu 18.04
Toolchain: Zephyr SDK version 0.11.2
tag: zephyr-v2.2.0

**Additional context**
N/A

--- a/include/posix/time.h
+++ b/include/posix/time.h
@@ -6,6 +6,11 @@
 #ifndef ZEPHYR_INCLUDE_POSIX_TIME_H_
 #define ZEPHYR_INCLUDE_POSIX_TIME_H_
 
+/* Read standard header.  This may find <posix/time.h> since they
+ * refer to the same file when include/posix is in the search path.
+ */
+#include <time.h>
+
 #ifdef CONFIG_NEWLIB_LIBC
 /* Kludge to support outdated newlib version as used in SDK 0.10 for Xtensa */
 #include <newlib.h>
@@ -101,4 +106,9 @@ int nanosleep(const struct timespec *rqtp, struct timespec *rmtp);
 #include <syscalls/time.h>
 #endif /* CONFIG_ARCH_POSIX */
 
+#else /* ZEPHYR_INCLUDE_POSIX_TIME_H_ */
+/* Read the toolchain header when <posix/time.h> finds itself on the
+ * first attempt.
+ */
+#include_next <time.h>
 #endif /* ZEPHYR_INCLUDE_POSIX_TIME_H_ */

--- a/samples/posix/gettimeofday/README.rst
+++ b/samples/posix/gettimeofday/README.rst
@@ -7,10 +7,11 @@ Overview
 ********
 
 This sample application demonstrates using the POSIX gettimeofday()
-function to display the absolute wall clock time every second. At
-system startup, the current time is queried using the SNTP networking
-protocol, enabled by setting the :option:`CONFIG_NET_CONFIG_CLOCK_SNTP_INIT`
-and :option:`CONFIG_NET_CONFIG_SNTP_INIT_SERVER` options.
+function to display the absolute wall clock time and local time every
+second. At system startup, the current time is queried using the SNTP
+networking protocol, enabled by setting the
+:option:`CONFIG_NET_CONFIG_CLOCK_SNTP_INIT` and
+:option:`CONFIG_NET_CONFIG_SNTP_INIT_SERVER` options.
 
 Requirements
 ************

--- a/samples/posix/gettimeofday/src/main.c
+++ b/samples/posix/gettimeofday/src/main.c
@@ -16,6 +16,9 @@ int main(void)
 
 	while (1) {
 		int res = gettimeofday(&tv, NULL);
+		time_t now = time(NULL);
+		struct tm tm;
+		localtime_r(&now, &tm);
 
 		if (res < 0) {
 			printf("Error in gettimeofday(): %d\n", errno);
@@ -23,8 +26,9 @@ int main(void)
 		}
 
 		printf("gettimeofday(): HI(tv_sec)=%d, LO(tv_sec)=%d, "
-		       "tv_usec=%d\n", (unsigned int)(tv.tv_sec >> 32),
-		       (unsigned int)tv.tv_sec, (unsigned int)tv.tv_usec);
+		       "tv_usec=%d\n\t%s\n", (unsigned int)(tv.tv_sec >> 32),
+		       (unsigned int)tv.tv_sec, (unsigned int)tv.tv_usec,
+		       asctime(&tm));
 		sleep(1);
 	}
 }

